### PR TITLE
Document the `metadata.query` method

### PR DESCRIPTION
**Pull request by vogelsgesang**

### CODE REVIEW

File: docs/api-ref.md [hunk 1]
```diff
@@ -1374,6 +1374,55 @@ Returns the `JobItem` requested.
...
+---
+
+## Metadata
+
+Using the `metadata` endpoint, one can use GraphQL to query metadata on, e.g., workbooks, datasources, the fields contained within them, and the relationships between them.
+
+### Metadata methods
+
+
+The Metadata methods are based upon the [Metadata API](https://help.tableau.com/current/api/metadata_api/en-us/index.html) REST methods.
```
Comment (t8y8) [inline docs/api-ref.md:1389]: They aren't really REST methods.

I would say something like "The Metadata methods call the Metadata API, which is a GraphQL interface" or something similar

File: docs/api-ref.md [hunk 2]
```diff
@@ -1374,6 +1374,55 @@ Returns the `JobItem` requested.
...
+Name | Description
+:--- | :---
+`query`          | The GraphQL query
+`variables`      | Variables for the GraphQL query
+`abort_on_error` | Whether to throw an exception on error
+
+
+**Returns**
+
+The query results as a structure of `dict`s and `list`s
```
Comment (t8y8) [inline docs/api-ref.md:1419]: The query results are json interpreted by the `json` package into dicts and lists, to be a little more explicit?

**Review by t8y8 — CHANGES_REQUESTED** [review]
Minor changes, but thanks for getting it started!

Docs folks should probably take a look too.

**Comment by vogelsgesang** [comment]
@t8y8 I addressed your issues in the new commit. Please double-check and let me know if there is anything else I should change

**Review by t8y8 — APPROVED** [review]
🚀

**Comment by bcantoni** [comment]
Thanks @vogelsgesang for the addition; merged.

**Pull request merged**